### PR TITLE
docs: reformat the state machine diagram using mermaid

### PR DESCRIPTION
**Pull request by tchaikov**
for better readability

### CODE REVIEW

**Comment by tchaikov** [comment]
before this change
- https://github.com/scylladb/scylladb/blob/master/docs/dev/topology-over-raft.md

after this change
- https://github.com/scylladb/scylladb/blob/fbe9cef098b2fe29055748c993445b4df50c7a71/docs/dev/topology-over-raft.md

File: docs/dev/topology-over-raft.md [hunk 1]
```diff
@@ -16,12 +16,16 @@ Node state can be one of those:
...
+```mermaid
+stateDiagram-v2
+    [*] --> bootstrapping|replacing
+    [*] --> [*]
+    bootstrapping|replacing --> normal
+    bootstrapping|replacing --> [*]
+    normal --> rebuilding
+    normal --> decommissioning|removing
+    rebuilding --> normal
+    decommissioning|removing --> [*]
```
Comment (avikivity) [inline docs/dev/topology-over-raft.md:28]: This should be "left", no? also "none" is not presented.

Comment (tchaikov) [reply docs/dev/topology-over-raft.md:28]: they are practically the "start" and "end" state. so I intentionally merge them into the start and end respectively. If you still think we should have them explicitly in the diagram, I will add them back.

Comment (tchaikov) [reply docs/dev/topology-over-raft.md]: i changed back to the named state. this preserves the naming of the original ascii-based diagram.

**Comment by tchaikov** [comment]
@kbr- and @gleb-cloudius could you please help review this change?

**Comment by tchaikov** [comment]
before this change
- https://github.com/scylladb/scylladb/blob/master/docs/dev/topology-over-raft.md

after this change
- https://github.com/scylladb/scylladb/blob/9839194151c297ed83855e66af260df5f7c6efab/docs/dev/topology-over-raft.md

**Comment by tchaikov** [comment]
@gleb-cloudius thanks! based on my reading of the code, i added some texts to the arcs. could you take another look?

**Comment by gleb-cloudius** [comment]
The arrows from `decommission|remove` to `left` and from `bootstrapping|replacing` to `normal` can be marked as operation succeeded. And from `bootstrapping|replacing` to `left` as operation failed.

**Comment by tchaikov** [comment]
before this change
- https://github.com/scylladb/scylladb/blob/master/docs/dev/topology-over-raft.md

after this change
- https://github.com/scylladb/scylladb/blob/b4c5d8ef319ec806769c84caf6f612189d7e0b31/docs/dev/topology-over-raft.md

**Comment by tchaikov** [comment]
@gleb-cloudius  thank you!updated per your suggestion.

**Review by gleb-cloudius — APPROVED** [review]

**Pull request closed without merging**